### ISSUE TITLE
Numeric data type when using Format String for sorting and conditional formatting

### ISSUE THREAD

**Issue by mybarbie9917** [otykier/TabularEditor#682]
I find Calculation Groups an extremely useful addition to Power BI, and I can't wait to use it more in my work. My reports have a lot of matrices and tables with a lot of conditional formattings like Font Color or Data Bar. The issues preventing me from switching from folders of clustered measure to a single table containing all of the time intelligence Calculation Groups are 
(1) my stakeholders use sorting all the time to look at numeric data in matrices, 
(2) they want to see green color for positive numbers and red color for negative numbers. 
Those requirements are very basic and can be achieved easily with multiple separate measures in PBI. 

I use Format String to get the needed format, but it also changes the numeric data type to string type, which is not desirable at all. I can't find a way to produce the same result of my current reports with Tabular Editor yet. It's very frustrating as I really want to incorporate such an amazing external tool into my work but unfortunately, it hasn't supported all formatting functions yet.

**Comment by otykier** [comment]
See [this comment](https://github.com/otykier/TabularEditor/issues/566#issuecomment-672664843). You likely have a calculation item in your model that causes all items to decay to the "Variant" data type, which causes Power BI Desktop to treat all measures as "Text". This is a bug in Power BI Desktop. You can work around it by removing the "Format" annotation as described in that comment.

**Comment by mybarbie9917** [comment]
Hi @otykier , 
Thanks for your answer. Unfortunately, I'm still facing the same issue with string data type. 

In TE, I chose the Model and put the below codes in the Advanced Scripting, then I clicked "Run Script". 
```
foreach(var m in Model.AllMeasures)
{
    m.RemoveAnnotation("Format");
}
```
I checked the Data Type of the *Name* item in the Calculation Groups and its data type is *String / Text* which can't be edited. I also didn't touch any of the **Format String* options. 

![2](https://user-images.githubusercontent.com/38081976/99611839-405e0180-29c9-11eb-8faa-f9ba2296e060.png)

In the below screenshot, I have 2 matrices. The above matrix is created with normal measures in PBI. The below matrix is created with the Calculation Groups from TE. The columns CY and PY are supposed to be numeric, so the users can sort them. In the below matrix, as I use the *Name* Calculation Group, which has default *Text* data type, the columns CY and PY are text, and hence can't be sorted, let alone be configured conditional formatting.  

![1](https://user-images.githubusercontent.com/38081976/99611794-2e7c5e80-29c9-11eb-8310-ae81caff0445.png)

When I tried to change the data type of the *Name* field to decimal number in PBI, I got this error message. 

![3](https://user-images.githubusercontent.com/38081976/99612874-5e2c6600-29cb-11eb-9eba-d1edcf8d0fdb.png)

In short, I don't know if I didn't follow your instructions correctly or the data type of all Calculation Groups items are *Text*. Is there a way to change the data type to numeric?  

Thank you.

**Comment by otykier** [comment]
There seems to be some kind of misunderstanding here. The [Name] column on the Calculation Group has to be "Text", because this is the column that contains the names of the calculation items ("CY", "CY WTD", "PY", etc.).

Since you're using the [TDV] measure in your matrix, and the calculation items do not seem to alter its format string, this is the measure you should consider for formatting.

For dynamic formatting with calculation items, you have to use a special technique with a hidden measure that defines the color. See #581 for more details on how to do this.

**Comment by mybarbie9917** [comment]
Thanks @otykier! I was able to set up conditional formatting with your trick using the hidden measure.  

One remaining issue is I want to use different columns in Calculation Groups for matrix sorting. My stakeholders use sorting all the time to find biggest number of current year and last year, biggest growth item, etc. Would it be possible to do interactively sorting? Thank you.  

[Sorting with Calculation Groups - Tabular Editor.zip](https://github.com/otykier/TabularEditor/files/5587011/Sorting.with.Calculation.Groups.-.Tabular.Editor.zip)

**Comment by otykier** [comment]
@mybarbie9917 sorry for the late reply. I'm not sure what you mean by "Would it be possible to do interactively sorting"? Could you please provide a detailed example, perhaps a screenshot, of what it is you'd like to achieve?
